nrf_modem/doc/bootloader.rst

```diff
@@ -0,0 +1,145 @@
+.. _nrf_modem_bootloader:
+
+Full firmware updates
+#####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Initializing the Modem library in bootloader mode gives access to the functionality of the modem bootloader, through the :file:`nrf_modem_bootloader.h` interface.
+The modem bootloader can be used to perform a full update of the modem firmware.
+Full modem firmware updates are the only operation that can be performed when the library is initialized in bootloader mode.
+AT commands and networking sockets are not available to the application when the modem is in bootloader mode.
+
+It is possible to switch between the bootloader and normal operation modes by reinitializing the library, by calling :c:func:`nrf_modem_shutdown` before :c:func:`nrf_modem_bootloader_init` or :c:func:`nrf_modem_init`, respectively.
+
+Memory requirements
+*******************
+
+The bootloader mode requires a shared memory area of size equal to the value of ``NRF_MODEM_SHMEM_BOOTLOADER_SIZE``.
+
+.. note::
+
+   When using |NCS|, the library is initialized by the glue, which will configure the size and position of the shared memory regions in RAM.
+
+Modem library initialization
+============================
+
+To perform a full firmware update of the modem, the library must be initialized in bootloader mode as shown in the following code:
+
+.. code-block:: c
+
+   /* Initialize modem to prepare for full modem firmware update */
+   nrf_modem_bootloader_init(bootloader_init_params);
+
+If the library has already been initialized in normal mode, it must be shut down through the :c:func:`nrf_modem_shutdown` function and reinitialized as shown in the following code:
+
+.. code-block:: c
+
+   nrf_modem_init(init_params);
+   /* Shutdown and re-initialize modem to prepare for full modem firmware update */
+   nrf_modem_shutdown();
+   nrf_modem_bootloader_init(bootloader_init_params);
+
+Considerations for corrupted modem firmware
+===========================================
+
+When designing your application, you might have to consider an interrupted modem firmware update process, which can lead to corrupted modem firmware.
+There can be various reasons, for example power loss, which can cause an interruption in a firmware update.
+The modem does not have a back-up ROM and hence, an interruption in the modem firmware update process can prevent the modem from further boot up.
+
+As a workaround to the above scenario, either the application must tolerate the situation or another means of recovery must be provided, for example, reprogramming the modem with PC tools.
+
+The Modem library must then be manually initialized with the following code:
+
+.. code-block:: c
+
```

Comment:
We could align the whole code block to the left similar to the other suggestions.